src/solvers/dg/1d/containers.jl

```diff
@@ -12,17 +11,56 @@ mutable struct ElementContainer1D{RealT<:Real, NVARS, POLYDEG} <: AbstractContai
   node_coordinates::Array{RealT, 3}    # [orientation, i, elements]
   surface_flux_values::Array{RealT, 3} # [variables, direction, elements]
   cell_ids::Vector{Int}                # [elements]
+  # internal `resize!`able storage
+  _u::Vector{RealT}
+  _u_t::Vector{RealT}
+  _u_tmp2::Vector{RealT}
+  _u_tmp3::Vector{RealT}
+  _node_coordinates::Vector{RealT}
+  _surface_flux_values::Vector{RealT}
 end
 
-function Base.copy!(dst::ElementContainer1D, src::ElementContainer1D)
-  dst.u                   = src.u
-  dst.u_t                 = src.u_t
-  dst.u_tmp2              = src.u_tmp2
-  dst.u_tmp3              = src.u_tmp3
-  dst.inverse_jacobian    = src.inverse_jacobian
-  dst.node_coordinates    = src.node_coordinates
-  dst.surface_flux_values = src.surface_flux_values
-  dst.cell_ids            = src.cell_ids
+# Only one-dimensional `Array`s are `resize!`able in Julia.
+# Hence, we use `Vector`s as internal storage and `resize!`
+# them whenever needed. Then, we reuse the same memory by
+# `unsafe_wrap`ping multi-dimensional `Array`s around the
+# internal storage.
+function Base.resize!(elements::ElementContainer1D{RealT, NVARS, POLYDEG},
+                      capacity) where {RealT, NVARS, POLYDEG}
+  n_nodes = POLYDEG + 1
+  @unpack _u, _u_t, _u_tmp2, _u_tmp3, _node_coordinates, _surface_flux_values,
+          inverse_jacobian, cell_ids = elements
+
+  resize!(_u, NVARS * n_nodes * capacity)
+  elements.u = unsafe_wrap(Array, pointer(_u),
+                           (NVARS, n_nodes, capacity))
+
+  resize!(_u_t, NVARS * n_nodes * capacity)
+  elements.u_t = unsafe_wrap(Array, pointer(_u_t),
+                             (NVARS, n_nodes, capacity))
+
+  resize!(_u_tmp2, NVARS * n_nodes * capacity)
+  _u_tmp2 .= zero(eltype(_u_tmp2))
+  elements.u_tmp2 = unsafe_wrap(Array, pointer(_u_tmp2),
+                                (NVARS, n_nodes, capacity))
+
+  resize!(_u_tmp3, NVARS * n_nodes * capacity)
+  _u_tmp3 .= zero(eltype(_u_tmp3))
```

Comment:
According to the [docs](https://docs.julialang.org/en/v1/base/collections/#Base.resize!), `resize!` does not modify existing values that still fit into the new size. Adhering to the principle of least astonishment, I'd thus prefer to not have a different behavior here. Or is there a reason why we can't zero these arrays manually after a resize (if it is necessary at all)?

Reply:
Same argument of course goes for the 2D version.

Reply:
Good catch. It's indeed unnecessary here since these temporary arrays are not used in Taal anymore and shall be removed when getting rid of the old code. I've removed the assignments.